docs/source-fabric/advanced/model_parallel/tp.rst

```diff
@@ -0,0 +1,280 @@
+##################
+Tensor Parallelism
+##################
+
+Tensor parallelism is a technique for training large models by distributing layers across multiple devices, improving memory management and efficiency by reducing inter-device communication.
```

Comment:
~~In addition to distributing layers, is it also distributing the matrix multiplication etc (or does it move the matrices to the same device when multiplying?~~ (NVM, you answer it below)

If it does the matrix multiplication in a distributed way, then a key point to mention here could be that this is useful when there are large layers (as opposed to a large model with lots of layers but small layers)